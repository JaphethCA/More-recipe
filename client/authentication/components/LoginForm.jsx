import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import TextField from '../../common/TextField';
import '../syles/authentication.scss';


/**
 * @description displays downvote button
 * @param {object} props - React props
 * @returns {ReactElement} markup
 */
const LoginForm = (props) => {
  const {
    validationErrors, serverErrors, onChange, onSubmit, formData
  } = props;
  return (
    <div className="row auth-form container">
      <div className="card col s12 m6 offset-m4 l4 offset-l4 z-depth-4">
        <div className="header-text">
          <h4>Log In</h4>
        </div>
        <div className="server-error-text">
          {serverErrors && <span > {serverErrors} </span> }
        </div>
        <form className="" onSubmit={onSubmit}>
          <div className="input-field col s12 m12">
            <TextField
              iconClassName="material-icons prefix"
              iconName="person"
              value={formData.email}
              onChange={onChange}
              name="email"
              labelClassName="active"
              label="Email"
              errorClass="error-class"
              errorText={validationErrors.email}
            />
          </div>
          <div className="input-field col s12 m12">
            <TextField
              iconClassName="material-icons prefix"
              iconName="lock"
              value={formData.password}
              onChange={onChange}
              name="password"
              type="password"
              labelClassName="active"
              label="Password"
              errorClass="error-class"
              errorText={validationErrors.password}
            />
          </div>
          <div className="input-field col s12">
            <button
              className="btn-large auth-btn"
              type="submit"
              disabled={props.isFetching}
            >
              {props.isFetching ? 'Loging In...' : 'Login' }
            </button>
          </div>
          <div className="auth-link">
            <span> Don&#39;t have an Account? <Link to="/signup" href="/signup"> Sign Up Here </Link></span>
          </div>
        </form>
      </div>
    </div>
  );
};


LoginForm.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  serverErrors: PropTypes.string.isRequired,
  validationErrors: PropTypes.objectOf(PropTypes.any).isRequired,
  formData: PropTypes.objectOf(PropTypes.any).isRequired,
  isFetching: PropTypes.bool.isRequired
};


export default LoginForm;
