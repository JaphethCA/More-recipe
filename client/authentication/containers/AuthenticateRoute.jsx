import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

const propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  history: PropTypes.objectOf(PropTypes.any).isRequired
};

export default (WrappedComponent) => {
  /**
   * @class Authenticate
   * @extends {Component}
   */
  class Authenticate extends Component {
    /**
     * @description react component life cyle function
     * @returns {undefined}
     */
    componentWillMount() {
      if (!this.props.isAuthenticated) {
        this.props.history.push('/landing');
      }
    }
    /**
 *
 * react component life cylce function
 * @memberof Authenticate
 * @returns {undefined}
 */
    componentWillUpdate() {
      if (!this.props.isAuthenticated) {
        this.props.history.push('/landing');
      }
    }
    /**
     *
     *
     * @returns {ReactElement} markup
     * @memberof Authenticate
     */
    render() {
      return <WrappedComponent {...this.props} />;
    }
  }

  Authenticate.propTypes = propTypes;

  const mapStateToProps = state => ({
    isAuthenticated: state.auth.isAuthenticated
  });

  return connect(mapStateToProps, { })(Authenticate);
};
