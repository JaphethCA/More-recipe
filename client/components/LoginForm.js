import React, { Component } from 'react'
import PropTypes from 'prop-types';
import {Redirect} from 'react-router-dom';
import classname from 'classnames';
import { signinValidator } from '../utils/validators';

class LoginForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      email: '',
      password: '',
      errors: '',
      redirect: false
    };

    this.onSubmit = this.onSubmit.bind(this);
    this.onChange = this.onChange.bind(this);
    this.isValid = this.isValid.bind(this);
  }
  
  onChange(e){
    this.setState({[e.target.name]: e.target.value})
  }

  isValid(){
    const {errors, isValid} = signinValidator(this.state);
    if (!isValid){
      this.setState({errors})
    }
    return isValid;
  }

  onSubmit(e){
    e.preventDefault();
    if (this.isValid()){
      this.setState({inputIsValid: true})
      this.props.handleLoginRequest(this.state).then(
        (res) => {this.setState({redirect: true})},
        (err) => this.setState({errors: err})
        );
    }
  }

  render() {
    const {errors, email, password} = this.state;
    if (this.state.redirect){
      return <Redirect to='/' />
    }
    return (
      <div className='row'>
        <form onSubmit={this.onSubmit} className='col m4 offset-m4'>
          <h3> Sign In </h3>
          <div className={classname('input-field')} >
              <input 
                type='text'
                onChange = {this.onChange}
                name='email'
              />
              <label htmlFor='email' > Email</label>
              {errors.email && <span> {errors.email[0]}</span>}
          </div>
          
          <div className={classname('input-field')} >
            <input 
              type='password'
              onChange = {this.onChange}
              name='password'
            />
            <label htmlFor='password' > Password</label>
            {errors.password && <span>{errors.password[0]}</span>}

          </div>

          <button type='submit' className='btn brown waves-effect waves-light'> Login </button>
          </form>
      </div>
    )
  }
}

LoginForm.propTypes = {
  handleLoginRequest: PropTypes.func.isRequired
}

export default LoginForm;