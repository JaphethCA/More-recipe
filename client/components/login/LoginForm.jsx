import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Redirect, Link } from 'react-router-dom';
import { signinValidator } from '../../utils/validators';
import './login_form.scss';


const propTypes = {
  handleLoginRequest: PropTypes.func.isRequired,
  auth: PropTypes.objectOf(PropTypes.any).isRequired
};


class LoginForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      serverErrors: this.props.auth.errors,
      validationErrors: '',
    };

    this.onSubmit = this.onSubmit.bind(this);
    this.onChange = this.onChange.bind(this);
    this.isValid = this.isValid.bind(this);
  }
  componentWillReceiveProps(nextProps) {
    if (nextProps.auth.errors !== null) {
      this.setState({ serverErrors: nextProps.auth.errors });
    }
  }
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  onSubmit(event) {
    event.preventDefault();
    if (this.isValid()) {
      this.props.handleLoginRequest(this.state);
    }
  }

  isValid() {
    const { errors, isValid } = signinValidator(this.state);
    if (!isValid) {
      this.setState({ validationErrors: errors });
    }
    return isValid;
  }

  render() {
    const { validationErrors } = this.state;

    if (this.props.auth.isAuthenticated) {
      window.location.reload();
      return <Redirect to="/" />;
    }
    return (
      <div className="row login-form">
        <div className="card-panel col s12 m6 offset-m3 l4 offset-l4 z-depth-4">
          <div className="auth-text">
            <h4>Log In</h4>
          </div>
          <div className="input-field s12 center">
            {this.state.serverErrors && <span className="red-text align-center" > {this.state.serverErrors} </span> }
          </div>

          <form className="" onSubmit={this.onSubmit}>
            <div className="row" >
              <div className="input-field col s12 m12">
                <i className="material-icons prefix">person</i>
                <input
                  type="text"
                  onChange={this.onChange}
                  name="email"
                  placeholder="Email Address"
                />
                <label htmlFor="email" className="active" > Email</label>
                {validationErrors.email && <span className="error-text"> { validationErrors.email[0] }</span>}
              </div>
            </div>

            <div className="row">
              <div className="input-field col s12 m12" >
                <i className="material-icons prefix">lock</i>
                <input
                  type="password"
                  onChange={this.onChange}
                  name="password"
                  placeholder="Enter Your Password"
                  className="validate"
                />
                <label htmlFor="password" className="active "> Password</label>
                {validationErrors.password && <span className="error-text"> { validationErrors.password[0]}</span>}
              </div>
            </div>

            <div className="row">
              <div className="input-field col s12">
                <button className="btn-large sign-up-btn" type="submit">Login
                </button>
              </div>
            </div>
            <div className="sign-up-link">
              <span> Have an Account? <Link to="/signup" href=" "> Sign Up Here </Link></span>
            </div>
          </form>
        </div>
      </div>
    );
  }
}

LoginForm.propTypes = propTypes;

export default LoginForm;
