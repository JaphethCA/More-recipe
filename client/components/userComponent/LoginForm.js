import React, { Component } from 'react'
import PropTypes from 'prop-types';
import {Redirect} from 'react-router-dom';
import classname from 'classnames';
import TextField from '../TextField';
import { signinValidator } from '../../utils/validators';

class LoginForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      email: '',
      password: '',
      errors: '',
      redirect: false
    };

    this.onSubmit = this.onSubmit.bind(this);
    this.onChange = this.onChange.bind(this);
    this.isValid = this.isValid.bind(this);
  }
  
  onChange(e){
    this.setState({[e.target.name]: e.target.value})
  }

  isValid(){
    const {errors, isValid} = signinValidator(this.state);
    if (!isValid){
      this.setState({errors})
    }
    return isValid;
  }

  onSubmit(e){
    e.preventDefault();
    if (this.isValid()){
      this.setState({inputIsValid: true})
      this.props.handleLoginRequest(this.state).then(
        (res) => {this.setState({redirect: true})},
        (err) => this.setState({errors: err})
        );
    }
  }

  render() {
    const {errors, email, password} = this.state;
    if (this.state.redirect){
      return <Redirect to='/' />
    }
    return (
      <div className='row'>
      <div className='card-panel col s12 m10 offset-m1 l6 offset-l3 z-depth-4'>
        <h4>Login Form</h4>
        <form className='' onSubmit={this.onSubmit}>
          <div className={classname('input-field','row',{'red-text':errors.email})} >
                <input 
                  className='col s12 m12'
                  type='text'
                  onChange = {this.onChange}
                  name='email'
                  />
                <label htmlFor='email' > Email</label>
                {errors.email && <span> {errors.email[0]}</span>}
            </div>
            <div className={classname('input-field','row',{'red-text':errors.password})} >
              <input 
                className='col s12 m12'
                type='password'
                onChange = {this.onChange}
                name='password'
                />
              <label htmlFor='password' > Password</label>
              {errors.password && <span>{errors.password[0]}</span>}
          </div>
          <div className="row">
              <div className="input-field col s12">
                <button className="btn brown waves-effect waves-light col m4 right" type="submit">Login
                </button>
            </div>
          </div>
        </form>
      </div>
      </div>
    )
  }
}

LoginForm.propTypes = {
  handleLoginRequest: PropTypes.func.isRequired
}

export default LoginForm;