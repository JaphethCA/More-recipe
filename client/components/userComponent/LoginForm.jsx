import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Redirect, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { addFlashMessage } from '../../actions/flashMessage';
import { signinValidator } from '../../utils/validators';
import '../../styles/sass/login_form.scss';

class LoginForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      errors: '',
      redirect: false,
      hasErrored: false
    };

    this.onSubmit = this.onSubmit.bind(this);
    this.onChange = this.onChange.bind(this);
    this.isValid = this.isValid.bind(this);
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  isValid() {
    const { errors, isValid } = signinValidator(this.state);
    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }

  onSubmit(e) {
    e.preventDefault();
    if (this.isValid()) {
      this.setState({ inputIsValid: true });
      this.props.handleLoginRequest(this.state).then(
        (res) => {
          this.setState({ redirect: true});
          location.reload();
        },
        (err) => {
          console.log(err.data);
          this.setState({ hasErrored: true });
        }
      );
    }
  }

  render() {
    const { 
      errors, email, password, hasErrored 
    } = this.state;
    if (this.state.redirect) {
      return <Redirect to="/" />;
    }
    return (

      <div className="row login-form">
        <div className="card-panel col s12 m6 offset-m3 l4 offset-l4 z-depth-4">
        <div className="input-field s12 center login-text">
          <h4>LOGIN FORM</h4>
          {hasErrored && <span className="red-text align-center" > Invalid Credentials</span> }
        </div>
        <form className="" onSubmit={this.onSubmit}>
          <div className="row" >
            <div className="input-field col s12 m12">
              <i className="material-icons prefix">person</i>
              <input
                  type="text"
                  onChange ={this.onChange}
                  name="email"
                  placeholder="Email Address"
                />
              <label htmlFor="email" className="active" > Email</label>
              {errors.email && <span className="error-text"> {errors.email[0]}</span>}
            </div>
          </div>
          <div className="row">
            <div className="input-field col s12 m12" >
              <i className="material-icons prefix">lock</i>
              <input
                  type="password"
                  onChange ={this.onChange}
                  name="password"
                  placeholder="Enter Your Password"
                />
              <label htmlFor="password" className="active" > Password</label>
              {errors.password && <span className="error-text">{errors.password[0]}</span>}
            </div>
          </div>

          <div className="row">
            <div className="input-field col s12">
                <button className="btn-large brown waves-effect waves-light col s12 m6 offset-m3 center" type="submit">Login
                </button>
              </div>
          </div>
          <div className="sign-up-link">
            <span>Don't have login? <Link to="/signup">Sign Up Here</Link></span>
          </div>
        </form>
      </div>
      </div>
    );
  }
}

LoginForm.propTypes = {
  handleLoginRequest: PropTypes.func.isRequired,
  addFlashMessage: PropTypes.func.isRequired
};

export default connect(null, { addFlashMessage })(LoginForm);
