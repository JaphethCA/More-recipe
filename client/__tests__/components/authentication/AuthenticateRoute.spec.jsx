import React from 'react';
import { shallow } from 'enzyme';
import AuthenticateRoute from '../../../authentication/containers/AuthenticateRoute';
import LoginForm from '../../../authentication/components/LoginForm';
import mockstore from '../../__mock__/configMockStore';

const store = mockstore({
  auth: {
    isAuthenticated: true
  }
});
const props = {
  isAuthenticated: false,
  history: {},
  setCurrentUser: jest.fn(),
  store
};

const state = {
  isAuthenticated: false
};

/**
 * @class MockComponent
 */
class MockComponent extends React.Component {
  /**
   * @returns {jsx} react element
   */
  render() {
    return (<div>Component</div>);
  }
}

const WrappedComponent = AuthenticateRoute(MockComponent);

describe('<AuthenticateRoute />', () => {
  it('should render without exploding', () => {
    const wrapper = shallow(<WrappedComponent {...props} />);
    expect(wrapper).toBeDefined();
    expect(wrapper.length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it(
    'should handle routes authentication when a user navigates to a new page',
    () => {
      const wrapper = shallow(<WrappedComponent {...props} {...state} />);
      const componentWillReceivePropsSpy = jest
        .spyOn(wrapper.instance(), 'componentWillReceiveProps');
      wrapper.setProps({ ...props, isAuthenticated: true });
      expect(componentWillReceivePropsSpy).toHaveBeenCalled();
      expect(toJson(wrapper)).toMatchSnapshot();
    }
  );
});
