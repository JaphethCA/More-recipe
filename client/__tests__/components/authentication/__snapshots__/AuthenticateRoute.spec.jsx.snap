// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthenticateRoute /> renders without exploding 1`] = `
<Authenticate
  history={Object {}}
  isAuthenticated={true}
  setCurrentUser={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`<AuthenticateRoute /> should receive new props 1`] = `
<Authenticate
  history={Object {}}
  isAuthenticated={true}
  setCurrentUser={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
